PSKNodeServer/PSKNodeServer/API/scanditScan/ScanditScannerViewController.swift

```diff
@@ -6,22 +6,78 @@
 //
 
 import UIKit
+import ScanditBarcodeCapture
 
 class ScanditScannerViewController: UIViewController {
     @IBOutlet weak var apiKeyLabel: UILabel!
     private let scanditApiKey: String
+    private let symbologies: [Symbology]
+    private let dataCaptureContext: DataCaptureContext
+    private var barcodeCapture: BarcodeCapture
     
     override func viewDidLoad() {
         super.viewDidLoad()
         apiKeyLabel.text = scanditApiKey
     }
     
+    override func viewWillAppear(_ animated: Bool) {
+        barcodeCapture.addListener(self)
+        
+        let cameraSettings = BarcodeCapture.recommendedCameraSettings
+        Camera.default?.apply(cameraSettings)
+        dataCaptureContext.setFrameSource(Camera.default)
+        Camera.default?.switch(toDesiredState: .on)
+        
+        let captureView = DataCaptureView(context: dataCaptureContext, frame: view.bounds)
+        captureView.autoresizingMask = [.flexibleWidth, .flexibleHeight]
+        view.addSubview(captureView)
+        
+        _ = BarcodeCaptureOverlay(barcodeCapture: barcodeCapture, view: captureView)
+    }
+    
     required init?(coder aDecoder: NSCoder) {
         fatalError("init(coder:) has not been implemented")
     }
     
-    init(with scanditApiKey: String){
+    init(with scanditApiKey: String, andWith symbologies: [Symbology]){
         self.scanditApiKey = scanditApiKey
+        self.symbologies = symbologies
+        self.dataCaptureContext = DataCaptureContext(licenseKey: scanditApiKey)
+        
+        let barcodeCaptureSettings = ScanditScannerViewController.createBarcodeCaptureSettings(with: symbologies)
+        self.barcodeCapture = BarcodeCapture(context: dataCaptureContext, settings: barcodeCaptureSettings)
+        
         super.init(nibName: "ScanditScannerViewController", bundle: nil)
     }
+    
+    private static func createBarcodeCaptureSettings(with symbologies: [Symbology]) -> BarcodeCaptureSettings {
+        let barcodeCaptureSettings = BarcodeCaptureSettings()
+        
+        let compositeTypes: CompositeType = [.a, .b, .c]
```

Comment:
Should we enable these composite types everytime? Isn't there any relation to selected symbologies?

Reply:
Yes, it is a separate thing from symbologies (even though symbologies included in composite code should be enabled this is just a switch to add composite codes support)...